lib/bundle/locker.rb

```diff
@@ -75,7 +75,7 @@ def lock(entries, global: false, file: nil, no_lock: false)
 
       json = JSON.pretty_generate(lock)
       begin
-        lockfile.unlink if lockfile.exist?
+        lockfile.unlink if lockfile.exist? && !lockfile.symlink?
```

Comment:
Feels a bit weird to just not unlink it if it's a symlink. Instead, how about resolving the symlink and changing `lockfile` to point to the destination file instead? Thoughts?

Reply:
Makes a lot of sense to me, I like that more than not deleting the file. Alternatively, we could just never unlink the file? It could break some workflows, but we wouldn't have to resolve any symlinks (not that it's hard to do).